### PR TITLE
feat: project deletion reminder and deletion of trashed objects.

### LINKED ISSUE #422: Scheduled jobs to delete the projects and associated objects after the cool-off period (30 days)

### LINKED ISSUE #423: Notify the project owners about the delete action via email
Once the project moved to trash, the project owner & creator would receive an email notifying them about the deletion.
User should also get reminder mail 1 week and 1 day before deleting the project permanently from storage.

### PR DESCRIPTION
**Pull request by NishaSharma14**
Fixes #423 
Fixes #422

### CODE REVIEW

**Comment by NishaSharma14** [comment]
image

**Review by CS76 — CHANGES_REQUESTED** [review]

File: resources/views/vendor/mail/project-archival.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+@component('mail::message')
+
+Dear nmrXiv user,
```
Comment (CS76) [inline resources/views/vendor/mail/project-archival.php:3]: Dear nmrXiv user,

Your nmrXiv project <project name> was flagged as archived on <date time(TZ)>.

A warning will now be displayed on your project, indicating that the project is no longer maintained.

You may unarchive your project if you would like to. To unarchive your project:
Visit the project page (link).
Select the project you want to unarchive, and click Project settings >  Unarchive.
In the confirmation dialogue, click confirm.

For further questions, please visit our documentation site or contact us at info@nmrxiv.org.

Regards,
The nmrXiv Team

File: resources/views/vendor/mail/project-deletion-failure.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+@component('mail::message')
+
+Dear admin,
+
+
```
Comment (CS76) [inline resources/views/vendor/mail/project-deletion-failure.php:5]: Project name - identifier and any other error-related details.

**Review by CS76 — CHANGES_REQUESTED** [review]

File: resources/views/vendor/mail/project-deletion.blade.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+{{ __('Your project - :project has been moved to trash successfully on :deletedOn. It will stay in trash until the cool-off period of 30 days after which it will be deleted permanently.', ['project' => $projectName, 'deletedOn' => $deletedOn]) }}
+
+You can always restore your project anytime before the cool-off period. 
+To restore:
+Go to your Trash section click on Project >> Project Settings >> Restore Project,  or click on the button below.
+
+
+If you do not recognize this action please report to us at info@nmrxiv.org.
+
```
[the remark this answers is not in the document]
Comment (NishaSharma14) [reply resources/views/vendor/mail/project-deletion.blade.php:13]: image

File: app/Actions/Project/DeleteProject.php [hunk 4]
```diff
@@ -32,8 +36,231 @@ public function delete($project)
...
+     */
+    public function deletePermanent($project)
+    {
+        if (! $project->is_public) {
+            try {
+                return DB::transaction(function () use ($project) {
+                    $this->deleteObjects($project);
+                });
+            } catch (Throwable $e) {
+                throw $e;
```
Comment (NishaSharma14) [inline app/Actions/Project/DeleteProject.php:60]: send notification first and then throw error

Comment (NishaSharma14) [reply app/Actions/Project/DeleteProject.php:60]: move cool-off period value to config(app.php)

File: resources/views/vendor/mail/project-archival.blade.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
+@component('mail::message')
+
+Dear nmrXiv user,
+
+{{ __('Your project - :project has been archived successfully.', ['project' => $projectName]) }}
+
+A warning will now be displayed on your project, indicating that the project is no longer maintained.
+
+You may unarchive your project if you would like to. To unarchive your project:
+Visit the project page (link).
```
Comment (NishaSharma14) [inline resources/views/vendor/mail/project-archival.blade.php:10]: Add message on separate lines for all mails

**Pull request merged**